samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -87,6 +92,7 @@ public final void init(Config config, TaskContext context) throws Exception {
 
     // create the operator impl DAG corresponding to the logical operator spec DAG
     this.operatorImplGraph = new OperatorImplGraph(streamGraph, config, context, clock);
+    this.eosDispatcher = EndOfStreamManager.createDispatcher(streamGraph);
```

Comment:
So, we create the eosDispatcher multiple times, one per task. Is it due to the fact that we are initializing and creating the streamGraph once per task instance? We can live with it for now. But should revisit this when we have a serialized DAG.